pkg/cmd/image/tag.go

```diff
@@ -59,17 +60,25 @@ func Tag(ctx context.Context, client *containerd.Client, options types.ImageTagO
 	}
 	defer done(ctx)
 
-	image, err := imageService.Get(ctx, srcName)
+	// Ensure all the layers are here: https://github.com/containerd/nerdctl/issues/3425
+	err = EnsureAllContent(ctx, client, srcName, options.GOptions)
+	if err != nil {
+		log.G(ctx).Warn("Unable to fetch missing layers before committing. " +
```

Comment:
On tag, we do not hard error if we fail to retrieve the missing layers.
This is to preserve the previous behavior, in case we are offline.
Warn the user instead, and point it to a special purpose ticket that provides explanations.